testflight/rename_resource_test.go

```diff
@@ -0,0 +1,34 @@
+package testflight_test
+
+import (
+	uuid "github.com/nu7hatch/gouuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+)
+
+var _ = Describe("Renaming a resource", func() {
+
+	It("preserved data", func() {
+		guid, err := uuid.NewV4()
+		Expect(err).ToNot(HaveOccurred())
+
+		setAndUnpausePipeline(
+			"fixtures/resource-with-params.yml",
+			"-v", "unique_version="+guid.String(),
+		)
+
+		fly("trigger-job", "-j", inPipeline("without-params"), "-w")
+		build := fly("builds", "-p", pipelineName)
+		Expect(build).To(gbytes.Say(pipelineName + "/without-params"))
+
+		setPipeline(
+			"fixtures/rename-resource-with-params.yml",
+			"-v", "unique_version="+guid.String(),
+		)
+
+		fly("trigger-job", "-j", inPipeline("without-params"), "-w")
+		build = fly("builds", "-p", pipelineName)
+		Expect(build).To(gbytes.Say(pipelineName + "/without-params"))
```

Comment:
Looks good, but maybe you could also test that it retains version history? e.g. something like

```go
resourceVersions := fly("resource-versions", "-r", inPipeline("new-git-resource"))
Expect(resourceVersions).To(gbytes.Say(originalGUID.String()))
```

In which case you'd also want to generate a new `guid` when you set the pipeline for the second time (that way, the original guid will only be a resource version if renaming the resource didn't clear the version history)

Reply:
Had to re-work that test, gave me the chance to learned about the mock resource!